fastlane/Fastfile

```diff
@@ -46,26 +69,17 @@ end
 desc 'Runs the unit tests for the project'
 lane :test do | options |
   run_tests(
-    workspace: WORKSPACE,
-    scheme: INTERNAL_SCHEME,
+    workspace: @WORKSPACE,
+    scheme: @INTERNAL_SCHEME,
     configuration: 'Debug',
     clean: true,
-    xcargs: SKIP_CODE_SIGNING_XCARGS
+    xcargs: @SKIP_CODE_SIGNING_XCARGS
   )
 end
 
 desc 'Builds the app for App Store distribution'
 lane :build_app_store do | options |
-  build_mac_app(
-    workspace: WORKSPACE,
-    scheme: APP_STORE_SCHEME,
-    configuration: 'Release',
-    clean: true,
-    output_directory: BUILD_FOLDER,
-    # Skip archive so we don't need to worry about code signing
-    skip_archive: true,
-    xcargs: SKIP_CODE_SIGNING_XCARGS
-  )
+  build_simplenote(codesign: false)
 end
```

Comment:
I made this lane for the CI step that ensures the App Store target builds. There are two different _targets_, actual targets, see #853.

That might not be the case for long, but we'll still need to have a clean option to make the Internal and App Store builds, regardless of whether they live in two targets or the same target but different build configurations.

Reply:
I found it a bit confusing that the lane was called `build_app_store` but that you would use `codesign: false`. Without any more context, I would think that this lane would be used by `build_and_upload_app_store` and would require codesign and would think this was a bug.

Maybe there can be a better name for this then, like `ensure_builds_for_appstore`? (not loving that one either tbh). Or make it more clear in the `desc` and via comments that this is intended only to test that the project just builds, but not intended to upload the generated build to app store?

Reply:
What about `test_app_store_build`? This one is confusing, too, because it doesn't actually run the tests. I'll go for this one and update if we can find a better alternative.

By the way, this should be a _temporary**™**_ check, just till we address https://github.com/Automattic/simplenote-macos/issues/853. That's to say, even if we don't settle on a perfect naming it's fine because it'll be gone "soon".

Reply:
Another option is to:
- get rid of `build_simplenote()` function and move its code to this lane directly
- Call this `build_app_store` lane from `build_and_upload_appstore` (instead of `build_simplenote`) then
- And just explicitly use `codesign:false` in the CI config at call site when the intent is only to test the build works but not upload it

Would avoid having to find names that could end up being confusing and instead make it clear at call site that codesign is disabled in CI when the job is about building without uploading